components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/context/AuthenticationContext.java

```diff
@@ -90,6 +90,7 @@
     private List<String> requestedAcr;
     private AcrRule acrRule = AcrRule.EXACT;
     private String selectedAcr;
+    private Map<String,  AuthenticatedIdPData> authenticatedIdPsPerApp = new HashMap<>();
```

Comment:
serialVersionUID of the class will need a change as this is modifying a serializable class.

Reply:
If we introduce a new uuid, it may break the existing sessions. Since it is not a major change, I think we can skip it. I handled this change gracefully.